### DISCUSSION TITLE
Git commit status for GitLab is not showing success or failure.

### DISCUSSION THREAD

**Discussion by Ayatallah** [fluxcd/flux2#1451] [Q&A]
I've a AKS k8s cluster with Flux installed and working perfectly fine as well as Notifications wokring perfectly fine with Rocket Provider and I was trying to add Git Commit Status for GitLab. I've followed doc [here]( https://fluxcd.io/docs/guides/notifications/#git-commit-status ) and did the following steps: 
1- Created Gitlab Personal Access Token with write access to my repository
2-Export GitLab token
```bash
export $GITLAB_TOKEN=<your-token>
```
3- Store the generated token in a secret in your cluster.
```bash
kubectl -n flux-system create secret generic gitlab --from-literal=token=$GITLAB_TOKEN
```

4- Create git provider whose address points to the same git repository as the event source originates from.
```bash
flux create alert-provider gitlab-provider \
  --type gitlab \
  --address <my-repo-link> \
  --secret-ref gitlab \
  --export | tee -a gitlab-notification.yaml
```

5- Create Namespace, with test deployment and kustomization (test) tracking the path where all objects in the created namespace resides. 

6- Create alert.
```bash
flux create alert gitlab-alert \
 --event-severity info \
 --event-source Kustomization/test \
 --provider-ref gitlab-provider \
 --export | tee -a gitlab-notification.yaml
```

7- Commit and push to my repo

8- Make change, create error or anything to see commit status

The problem is no any success or failure status show within my commits. The owner account of the personal access token just keeps on creating pipeline with every commit but the pipeline get status "created" and duration "in progress" and i never get to see status change. As in the following snapshot: I'm not sure what could I be missing in the above steps.

<img width="1137" alt="Screen Shot 2021-05-26 at 6 48 53 PM" src="https://user-images.githubusercontent.com/10837400/119700723-dc266980-be53-11eb-86a0-00523217fbb4.png">
and
<img width="965" alt="Screen Shot 2021-05-26 at 6 43 58 PM" src="https://user-images.githubusercontent.com/10837400/119701752-0167a780-be55-11eb-8dcd-95aa2d43339a.png">

**Comment by Ayatallah** [comment]
@phillebaba @stefanprodan your help here would be much appreciated!

**Comment by janfuhrer** [comment]
for me it works only with full `api` (read/write) access for the access-token.

You can see an error in the logs of the notification-controller:
`kubectl logs -n flux-system notification-controller-xxxxxxxxxx-xxxxx`

Something like this:
```bash
{"level":"error","ts":"2021-11-24T07:52:11.145Z","logger":"event-server","msg":"failed to send notification","reconciler kind":"Kustomization","name":"<kustomization-name>","namespace":"test","error":"POST https://<gitlab-url>/api/v4/projects/<user>/<repository>/statuses/<commit>: 403 {error: insufficient_scope}, {error_description: The request requires higher privileges than provided by the access token.}, {scope: api read_api}"}
```

**Comment by somtochiama** [comment]
@Ayatallah Could you paste logs from the notification controller after it receives the events?
